### PR TITLE
Gunsmithing

### PR DESCRIPTION
**Pull request by lordofhell666**
#### Summary

SUMMARY: Content - "Adds new gunsmithing tools such as shop press, induction annealer, lathe and advanced hand loading kit with various recipes to make better bullets"

#### Purpose of change

Guns are currently very underdeveloped. This is the first part of a multistage effort to improve their usability and realism. As someone who has been shooting since I was 9, pistols being unable to shoot across the street and high quality handloaded bullets being absent feels incredibly weird.

#### Describe the solution

This addition works by allowing the player to create superior home loaded bullets with the use of specialized equipment instead of the rather improv-like nature of the current handload system. The current system still remains, I've simply added this as an alternative for those who want to put in the effort to get better, replenishable bullets.

#### Testing

Played a gun-focused character for several hours, using debug menu to cheat myself in the materials required to craft everything. Recipes all work, items all obtainable. Guns still require care - firing one in the middle of a zombie infested city is still nigh suicidal unless you're Rambo, but they no longer feel directly inferior to smacking everyone in the face with a fire axe.

#### Additional context

I've done handloading for many years in real life. The current system makes enough sense - without clean casings and purpose-shaped projectiles made with caliber sleeves and OAL gauges, it's fair enough the reloaded bullets would be questionable. All I've done is give the player a way to make those specialized cases and projectiles, and solved the issue of certain gun magazines being the rarest item in the wastes. This is the first stage in my plan to make gunsmithing (and guns in general) more fleshed out, the "custom" bullets were given their own folder to avoid clutter in the future when more custom bullets are added. Will have to rebalance the amount of ammo the player finds from zombies, as getting fifty 9x19mm JHP off a single zombie is kinda funny, but I doubt anyone would just carry them around like that.

In addition, the lathe, shop press, and induction annealer add more possibilities for future, non-gun related recipes.

### CODE REVIEW

File: data/json/recipes/recipe_others.json [hunk 1]
```diff
@@ -2229,5 +2229,162 @@
...
+      [ [ "cable", 4 ] ],
+      [ [ "power_supply", 1 ] ],
+      [ [ "plastic_chunk", 2 ] ],
+	  [ [ "circuit", 1 ] ]
+	]
+  },
+  {
+    "type": "recipe",
+    "activity_level": "MODERATE_EXERCISE",
+    "result": "lathe",
```
Comment (Rail-Runner) [inline data/json/recipes/recipe_others.json:2323]: This probably shouldn't be craftable *that* easily. That said, it's not impossible to build a lathe starting with hand tools, but it's extremely complicated; if you want more details, you could look up the series of books by David Gingery named "Build Your Own Metal Working Shop from Scrap", but I can tell that you'd need aluminum casting tools for this project, for a start.

File: data/json/items/resources/metal.json [hunk 2]
```diff
@@ -272,6 +272,46 @@
...
+    "name": { "str": "lump of brass", "str_pl": "lumps of brass" },
+    "description": "A heavy formed piece of brass.  Useful for some crafting recipes.",
+    "weight": "1000 g",
+    "volume": "250 ml",
+    "price": 2000,
+    "price_postapoc": 50,
+    "bashing": 12,
+    "to_hit": -4,
+    "stack_size": 2,
+    "//": "Density 7.60g/cm³ ~ 1.9kg/250ml @ 1000g/unit ~ stack 2 and formed as a solid ingot so 100% of this",
```
Comment (Rail-Runner) [inline data/json/items/resources/metal.json:288]: This comment is wrong; brass density is actually 8.73 g/cm^3.

File: data/json/recipes/recipe_others.json [hunk 3]
```diff
@@ -2229,5 +2229,162 @@
...
+      [ "reference_fabrication1", 1 ]
+    ],
+	"using": [ [ "forging_standard", 2 ] ],
+    "qualities": [
+      { "id": "HAMMER", "level": 2 },
+      { "id": "SAW_M", "level": 1 },
+      { "id": "SCREW", "level": 1 },
+      { "id": "WRENCH", "level": 1 }
+    ],
+    "tools": [ [ [ "welder", 400 ], [ "welder_crude", 300 ], [ "toolset", 300 ], [ "soldering_iron", 300 ], [ "oxy_torch", 700 ] ] ],
```
Comment (SirPendrak) [inline data/json/recipes/recipe_others.json:2341]: Correct me if i am wrong, but you would need more charges for using factory-grade welder than for some handmade one?

File: data/json/recipes/recipe_others.json [hunk 4]
```diff
@@ -2229,5 +2229,162 @@
...
+      [ "recipe_bullets", 1 ]
+    ],
+	"using": [ [ "forging_standard", 2 ] ],
+    "qualities": [
+      { "id": "HAMMER", "level": 2 },
+      { "id": "SAW_M", "level": 1 },
+      { "id": "SCREW", "level": 1 },
+      { "id": "WRENCH", "level": 1 }
+    ],
+    "tools": [ [ [ "welder", 100 ], [ "welder_crude", 300 ], [ "toolset", 300 ], [ "soldering_iron", 300 ], [ "oxy_torch", 100 ] ] ],
```
Comment (SirPendrak) [inline data/json/recipes/recipe_others.json:2274]: Usually crude welders and toolset uses x1,5 energy, not x3

**Comment by kholat** [comment]
I feel that when the big K sees this there is going to be a big s***show, however it is a valiant attempt I suppose.

**Comment by Maddremor** [comment]
The whole compressed range thing is kind of necessary due to the reality bubble. If all guns were using the same linear scale as buildings, most (if not all) would cover the bubble and thus have the same effective range.

File: data/json/recipes/recipe_others.json [hunk 5]
```diff
@@ -2229,5 +2229,162 @@
...
+  },
+  {
+    "type": "recipe",
+    "activity_level": "MODERATE_EXERCISE",
+    "result": "lathe",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_TOOLS",
+    "skill_used": "fabrication",
+    "difficulty": 4,
+    "time": "20 h",
```
Comment (mlangsdorf) [inline data/json/recipes/recipe_others.json:2328]: Hahaha no.  Have you seen the number of castings you have to do in Gingerry's "The Metal Lathe" plus the precision you need to level the bed?  It's a multi-month project, not the kind of thing you knock out in a long weekend.

**Review by mlangsdorf — COMMENTED** [review]
I'd like to see this split into 2 or 3 PRs:
* 1 for the irrelevant gun range changes that are going to get NAK'd.
* 1 for the new tools like the metal lathe and annealer
* 1 that depends on the new tools PR that adds the new bullet recipes

**Comment by kevingranade** [comment]
What @mlangsdorf said, but with more emphasis, you're doing too many things here, some if which are either not going to happen or are going to require significant as not back-and-forth to sort out.

Please reopen in the chunks he outlined.

**Pull request closed without merging**